From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Levy <tomlevy93@gmail.com>
Date: Sun, 5 Sep 2021 05:38:27 +1200
Subject: Force Composer 1 to be installed in the dockerfiles

The default version is now Composer 2, which is incompatible with the
current lock file.

Upgrading to Composer 2 requires running "composer update", which will
be done later.
---
 docker/domserver/Dockerfile        | 4 ++--
 docker/domserver/Dockerfile.source | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/docker/domserver/Dockerfile b/docker/domserver/Dockerfile
index b0880704a..ba388cd10 100644
--- a/docker/domserver/Dockerfile
+++ b/docker/domserver/Dockerfile
@@ -28,7 +28,7 @@ RUN useradd -m domjudge
 
 # Install composer
 ADD https://getcomposer.org/installer composer-setup.php
-RUN php composer-setup.php \
+RUN php composer-setup.php --1 \
     && mv /composer.phar /usr/local/bin/composer
 
 # Add DOMjudge source code
@@ -69,7 +69,7 @@ RUN useradd -m domjudge
 
 # Install composer
 ADD https://getcomposer.org/installer composer-setup.php
-RUN php composer-setup.php \
+RUN php composer-setup.php --1 \
     && mv /composer.phar /usr/local/bin/composer
 
 # Copy domserver directory and add script files
diff --git a/docker/domserver/Dockerfile.source b/docker/domserver/Dockerfile.source
index 4846c1af0..7660a4f13 100644
--- a/docker/domserver/Dockerfile.source
+++ b/docker/domserver/Dockerfile.source
@@ -45,7 +45,7 @@ RUN useradd -m domjudge
 
 # Install composer
 ADD https://getcomposer.org/installer composer-setup.php
-RUN php composer-setup.php \
+RUN php composer-setup.php --1 \
     && mv /composer.phar /usr/local/bin/composer
 
 # Install PHP dependencies
@@ -94,7 +94,7 @@ RUN useradd -m domjudge
 
 # Install composer
 ADD https://getcomposer.org/installer composer-setup.php
-RUN php composer-setup.php \
+RUN php composer-setup.php --1 \
     && mv /composer.phar /usr/local/bin/composer
 
 # Copy domserver directory and add script files
-- 
2.20.1

